static/js/components/dashboard/CourseDescription.js

```diff
@@ -10,58 +10,72 @@ import {
   STATUS_NOT_PASSED,
   STATUS_ENROLLED_NOT_VERIFIED,
   STATUS_VERIFIED_NOT_COMPLETED,
+  STATUS_OFFERED_NOT_ENROLLED,
   DASHBOARD_FORMAT,
 } from '../../constants';
 
 export default class CourseDescription extends React.Component {
   props: {
-    course: Course,
-    now: moment$Moment,
+    course: Course
   };
 
-  render() {
-    const { course, now } = this.props;
-    let text = "", enrolled = "";
-    let firstRun: CourseRun = {};
-    if (course.runs.length > 0) {
-      firstRun = course.runs[0];
+  courseDate(label: string, date: moment$Moment): string {
+    let formattedDate = date.format(DASHBOARD_FORMAT);
+    return `${label}: ${formattedDate}`;
+  }
+
+  courseDateMessage(courseStatus: string, firstRun: CourseRun): string {
+    if (!firstRun) {
+      return '';
     }
 
-    switch (course.status) {
+    let text = "";
+
+    switch (courseStatus) {
     case STATUS_PASSED:
-      text = "Complete!";
-      break;
-    case STATUS_ENROLLED_NOT_VERIFIED:
-      text = <span>
-        You need to upgrade to the Verified course to get MicroMasters credit
-        <IconButton name="help" colored/>
-      </span>;
-      enrolled = "(enrolled)";
+      if (firstRun.course_end_date) {
+        let courseEndDate = moment(firstRun.course_end_date);
+        text = this.courseDate('Ended', courseEndDate);
+      }
       break;
     case STATUS_NOT_OFFERED:
-      if (firstRun.status === STATUS_NOT_PASSED) {
-        text = 'You failed this course';
+      if (firstRun.status === STATUS_NOT_PASSED && firstRun.course_end_date) {
+        let courseEndDate = moment(firstRun.course_end_date);
+        text = this.courseDate('Ended', courseEndDate);
+      } else if (!_.isNil(firstRun.fuzzy_start_date)) {
+        text = `Coming ${firstRun.fuzzy_start_date}`;
       }
       break;
+    case STATUS_ENROLLED_NOT_VERIFIED:
```

Comment:
@gsidebo these 3 consecutive statements are equal to 
```js
if (course.status === STATUS_ENROLLED_NOT_VERIFIED || course.status === STATUS_VERIFIED_NOT_COMPLETED || course.status === STATUS_OFFERED_NOT_ENROLLED )  
** same functionality**
```

Which mean these 3 case have same functionality, please see http://www.w3schools.com/js/js_switch.asp